packages/share_plus/share_plus/lib/src/share_plus_linux.dart

```diff
@@ -35,7 +40,11 @@ class SharePlusLinuxPlugin extends SharePlatform {
           .join('&'),
     );
 
-    await launchUrl(uri);
+    if (await urlLauncher.canLaunch(uri.toString())) {
+      await urlLauncher.launchUrl(uri.toString(), const LaunchOptions());
+    } else {
+      throw Exception('Unable to share on web');
+    }
   }
```

Comment:
FYI: changing to this construction will break functionality, and has no advantages; see https://pub.dev/packages/url_launcher#checking-supported-schemes

Reply:
So, it's better to have only `launchUrl` in try catch block?

Reply:
[Checking the return of launchUrl](https://pub.dev/packages/url_launcher#checking-supported-schemes:~:text=if%20(!await%20launchUrl(uri))%20%7B%0A%20%20throw%20%27Could%20not%20launch%20%24uri%27%3B%0A%7D) is better construct?

Reply:
> So, it's better to have only `launchUrl` in try catch block?

I'm not sure what the goal of a try/catch would be given that you are currently throwing.

> [Checking the return of launchUrl](https://pub.dev/packages/url_launcher#checking-supported-schemes:~:text=if%20(!await%20launchUrl(uri))%20%7B%0A%20%20throw%20%27Could%20not%20launch%20%24uri%27%3B%0A%7D) is better construct?

If you don't have a specific need to know in advance of trying (which is rare, and certainly isn't the case given the code here), checking what actually happened is better than using a potentially inaccurate prediction of what might happen, yes.

Reply:
@miquelbeltran FYI

Reply:
If anyone can send a PR with the fix, I can review and merge asap

Reply:
Yeah I can do that.